tests/marqo_test.py

```diff
@@ -248,32 +321,48 @@ def create_test_index(
             self,
             cloud_test_index_to_use: Union[CloudTestIndex, None], open_source_test_index_name: Union[str, None],
             open_source_index_settings_dict: dict = None, open_source_index_kwargs: dict = None,
+            delete_index_documents_before_test: bool = True
     ):
         """Determines whether the test is executed in a cloud environment or within an open-source setup.
         If it's running in an open-source environment,
         this function creates the corresponding index and returns its name.
 
         In the case of cloud testing, it provides the name of the cloud index to be used.
         Additionally, it applies a unique run identifier to the index name and performs
-        cleanup operations on documents associated with the index.
+        cleanup operations on documents associated with the index if 'delete_index_documents_before_test' is True.
+
+        Args:
+            cloud_test_index_to_use (Union[CloudTestIndex, None]): The cloud test index to use in cloud environments.
+                If None, an error is raised.
+            open_source_test_index_name (Union[str, None]): The name of the open-source test index to create in
+                open-source environments. If None, no open-source index is created.
+            open_source_index_settings_dict (dict, optional): Additional settings to apply when creating
+                the open-source test index.
+            open_source_index_kwargs (dict, optional): Additional keyword arguments to pass when creating
+                the open-source test index.
+            delete_index_documents_before_test (bool, optional): If True, existing documents in the index will
+                be deleted before preparing it for testing. Default is True. Used only for cloud testing.
 
         Returns:
             The name of the index to be used, depending on the testing environment.
 
+        Raises:
+            ValueError: If 'cloud_test_index_to_use' is None in cloud environments.
+
         """
         client = marqo.Client(**self.client_settings)
         if client.config.is_marqo_cloud:
             if cloud_test_index_to_use is None:
                 raise ValueError("cloud_test_index_to_use must be specified for cloud tests")
             index_name_to_return = f"{cloud_test_index_to_use.value}-{self.index_suffix}"
-            self.prepare_cloud_index_for_test(index_name_to_return)
+            self.prepare_cloud_index_for_test(index_name_to_return, delete_index_documents_before_test)
```

Comment:
Can you break up your run on sentence into shorter simpler sentences? It will be easier to follow then.

Reply:
It's not necessarily an issue limited to tests; it can also occur in other scenarios. For example, when using the with_documents decorator, it initially calls create_test_index and then adds documents to it.
After that, another function is called, which also invokes create_test_index to retrieve the index name it should point to.
If we perform a cleanup operation at this point, any documents created within the with_documents decorator will be deleted.